crypto/keyring/keyring.go

```diff
@@ -775,65 +766,69 @@ func newRealPrompt(dir string, buf io.Reader) func(string) (string, error) {
 	}
 }
 
-func (ks keystore) writeLocalKey(name string, priv types.PrivKey, algo hd.PubKeyType) (Info, error) {
-	// encrypt private key using keyring
-	pub := priv.PubKey()
-	info := newLocalInfo(name, pub, string(legacy.Cdc.MustMarshal(priv)), algo)
-	if err := ks.writeInfo(info); err != nil {
+func (ks keystore) writeLocalKey(name string, privKey types.PrivKey) (*Record, error) {
+	k, err := NewLocalRecord(name, privKey, privKey.PubKey())
+	if err != nil {
 		return nil, err
 	}
 
-	return info, nil
+	return k, ks.writeRecord(k)
 }
 
-func (ks keystore) writeInfo(info Info) error {
-	key := infoKeyBz(info.GetName())
-	serializedInfo := marshalInfo(info)
+func (ks keystore) writeRecord(k *Record) error {
```

Comment:
let's add a go doc about the proper conditions for using this functions (eg: that the key must not exists).